zipline/assets/asset_writer.py

```diff
@@ -268,6 +344,21 @@ def _write_futures(self, futures, bind):
     def _write_equities(self, equities, bind):
         self._write_assets(equities, self.equities, 'equity', bind)
 
+    def check_for_tables(self, engine):
+        """
+        Checks if any tables are present in the current assets database.
+
+        Returns
+        -------
+        bool
+            True if any tables are present, otherwise False.
+        """
+        conn = engine.connect()
+        for table_name in table_names:
```

Comment:
I think lines 357-360 are the equivalent of doing the following:
```python
return engine.dialect.has_table(conn, table_names[0])
```
This is because if *any* of the tables have been created, it implies that they all have been. This is because `init_db` is called within an SQL transaction.